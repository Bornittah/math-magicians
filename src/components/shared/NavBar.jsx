import React from 'react';
import { Link } from 'react-router-dom';

const NavBar = () => (
  <header>
    <nav className="flex space-btn">
      <div className="logo">
        <Link to="/">Math Magician</Link>
      </div>
      <ul className="navbar flex space-btn">
        <li className="nav-link border-right">
          <Link to="/" activeClassName="link">Home</Link>
        </li>
        <li className="nav-link border-right">
          <Link to="/calculator" activeClassName="link">Calculator</Link>
        </li>
        <li className="nav-link">
          <Link to="/quotes" activeClassName="link" exact>Quotes</Link>
        </li>
      </ul>
    </nav>
  </header>
);

export default NavBar;
