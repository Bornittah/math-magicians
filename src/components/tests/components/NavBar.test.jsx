import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import NavBar from '../../shared/NavBar';

describe('Testing NavBar component', () => {
  it('Testing with snapshot', () => {
    const tree = renderer.create(
      <Router><NavBar /></Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
