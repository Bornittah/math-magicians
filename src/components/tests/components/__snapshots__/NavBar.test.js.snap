// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NavBar component Testing with snapshot 1`] = `
<header>
  <nav
    className="flex space-btn"
  >
    <div
      className="logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Math Magician
      </a>
    </div>
    <ul
      className="navbar flex space-btn"
    >
      <li
        className="nav-link border-right"
      >
        <a
          activeClassName="link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-link border-right"
      >
        <a
          activeClassName="link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="nav-link"
      >
        <a
          activeClassName="link"
          exact={true}
          href="/quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
